lua/gluatest/stubs/arg.lua

```diff
@@ -0,0 +1,197 @@
+local argMeta = {
+    __tostring = function( self )
+        return "Arg Matcher: " .. self.name
+    end,
+}
+
+--- Creates a new argument matcher
+--- @param name string The name of the matcher (for use in error messages)
+--- @param check fun(any): boolean A function that returns true if the argument matches
+local function New( name, check )
+    --- @class GLuaTest_StubArg
+    local arg = {
+        --- @type string
+        --- The name of the matcher, used in error messages
+        name = name,
+
+        --- A function that returns true if the argument matches
+        check = check,
+    }
+
+    return setmetatable( arg, argMeta )
+end
+
+--- @alias GLuaTest_StubArg_Matcher fun(any): boolean
+
+--- @class GLuaTest_StubArgs
+local Arg = {
+    --- The metatable for all argument matchers
+    argMeta = argMeta,
+
+    New = New,
+
+    --- Checks if the given argument is an argument matcher
+    --- @param arg any The argument to check
+    IsArgMatcher = function( arg )
+        return getmetatable( arg ) == argMeta
+    end,
+
+    --- Matches any argument
+    Any = New( "Any", function( _ )
+        return true
+    end ),
+
+    --- Matches nil
+    Nil = New( "Nil", function( arg )
+        return arg == nil
+    end ),
+
+    --- Matches any boolean
+    Boolean = New( "Any Boolean", function( arg )
+        return arg == true or arg == false
+    end ),
+
+    --- Matches any number
+    Number = New( "Any Number", function( arg )
+        return type( arg ) == "number"
+    end ),
+
+    --- Matches any string
+    String = New( "Any String", function( arg )
+        return type( arg ) == "string"
+    end ),
+
+    --- Matches any table
+    Table = New( "Any Table", function( arg )
+        return type( arg ) == "table"
+    end ),
+
+    --- Matches any function
+    Function = New( "Any Function", function( arg )
+        return type( arg ) == "function"
+    end ),
+
+    --- Matches any userdata
+    Userdata = New( "Any Userdata", function( arg )
+        return type( arg ) == "userdata"
+    end ),
+
+    --- Matches any thread
+    Thread = New( "Any Thread", function( arg )
+        return type( arg ) == "thread"
+    end ),
+
+    -- GMod Types
+
+    --- Matches any Vector
+    Vector = New( "Any Vector", function( arg )
+        return type( arg ) == "Vector"
+    end ),
+
+    --- Matches any Angle
+    Angle = New( "Any Angle", function( arg )
+        return type( arg ) == "Angle"
+    end ),
+
+    --- Matches any Color
+    Color = New( "Any Color", function( arg )
+        return debug.getmetatable( arg ) == FindMetaTable( "Color" )
```

Comment:
You might want to use `IsColor` instead.
https://gmodwiki.com/Global.IsColor

https://github.com/Facepunch/garrysmod/blob/master/garrysmod/lua/includes/util/color.lua#L36-L40

The bug https://github.com/Facepunch/garrysmod-issues/issues/2407 has been resolved.